test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -345,7 +344,11 @@ TEST_F(FileSystemImplTest, ExistingReadOnlyFileAndWrite) {
     std::string data(" new data");
     const Api::IoCallSizeResult result = file->write(data);
     EXPECT_TRUE(result.rc_ < 0);
-    EXPECT_EQ(result.err_->getErrorDetails(), "Bad file descriptor");
+#ifdef WIN32
+    EXPECT_EQ(IoFileError::IoErrorCode::Permission, result.err_->getErrorCode());
+#else
+    EXPECT_EQ(IoFileError::IoErrorCode::BadFd, result.err_->getErrorCode());
+#endif
```

Comment:
I'm wondering if we want to ensure these have the same observed error in the windows and unix implementations?

Reply:
This is my initial goal but the thing that kept from doing it is that the error codes that the platform codes are so different. 

The error in POSIX is `EBADF` which is a really generic error so it's hard to map it back to the more specific error. Alternatively, we could make the Windows error more generic but then we would be "losing" information.